src/definition/email/IEmailDescriptor.ts

```diff
@@ -0,0 +1,12 @@
+
+export interface IEmailDescriptor {
+    from?: string | undefined;
+    to?: string | Array<string> | undefined;
```

Comment:
Is the last `undefined` needed? I haven't touched TS in a while but I always thought having the question mark automatically added that

Reply:
The `?` and `undefined` do have different meanings: adding `?` makes it optional to _include_ that property when creating an object that follows an interface; while `undefined` allows for the inclusion of the property with `undefined` as a value.

So, if we removed the `? ` from the type, it would be mandatory to include `{ to: undefined }` at least in the object. On the other hand, if we removed `undefined`, _it wouldn't be possible to pass pass that property as `undefined`_, i.e. `{ to: undefined }`.

But we did mostly copy it from here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/meteor/email.d.ts#L4-L12 :joy:

Reply:
Makes sense to me. Been so long I've forgot about all the inner workings and specifics of TypeScript 🙈